src/EditorFeatures/CSharpTest/MisplacedUsings/MisplacedUsingsCodeFixProviderTests.cs

```diff
@@ -0,0 +1,1295 @@
+        /// Verifies that valid using statements in a namespace does not produce any diagnostics.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInNamespace_ValidUsingStatements()
+        {
+            var testCode = @"namespace TestNamespace
+{
+    using System;
+    using System.Threading;
+}
+";
+
+            return VerifyAnalyzerAsync(testCode, s_insidePreferenceOption, DiagnosticResult.EmptyDiagnosticResults);
+        }
+
+        /// <summary>
+        /// Verifies that having using statements in the compilation unit will not produce any diagnostics when there are type definition present.
+        /// </summary>
+        /// <param name="typeDefinition">The type definition to test.</param>
+        [Theory]
+        [InlineData(ClassDefinition)]
+        [InlineData(StructDefinition)]
+        [InlineData(InterfaceDefinition)]
+        [InlineData(EnumDefinition)]
+        [InlineData(DelegateDefinition)]
+        public Task WhenInsidePreferred_UsingsInCompilationUnitWithTypeDefinition_ValidUsingStatements(string typeDefinition)
+        {
+            var testCode = $@"using System;
+
+{typeDefinition}
+";
+
+            return VerifyAnalyzerAsync(testCode, s_insidePreferenceOption, DiagnosticResult.EmptyDiagnosticResults);
+        }
+
+        /// <summary>
+        /// Verifies that having using statements in the compilation unit will not produce any diagnostics when there are attributes present.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnitWithAttributes_ValidUsingStatements()
+        {
+            var testCode = @"using System.Reflection;
+
+[assembly: AssemblyVersion(""1.0.0.0"")]
+
+namespace TestNamespace
+{
+    using System;
+    using System.Threading;
+}
+";
+
+            return VerifyAnalyzerAsync(testCode, s_insidePreferenceOption, DiagnosticResult.EmptyDiagnosticResults);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix will properly reorder using statements when placing System first.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnit_UsingsMovedWithSystemPlacedFirst()
+        {
+            var testCode = @"using Microsoft.CodeAnalysis;
+using SystemAction = System.Action;
+using static System.Math;
+using System;
+
+using static System.String;
+using MyFunc = System.Func<int,bool>;
+
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Foo
+{
+    public class Bar
+    {
+    }
+}
+";
+
+            var fixedTestCode = @"namespace Foo
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using Microsoft.CodeAnalysis;
+    using static System.Math;
+    using static System.String;
+    using MyFunc = System.Func<int,bool>;
+    using SystemAction = System.Action;
+
+    public class Bar
+    {
+    }
+}
+";
+
+            DiagnosticResult[] expected =
```

Comment:
var

Reply:
➡️ Would result in a compile error here.

Reply:
Why?

Reply:
because it's using an array initializer instead of `new[]`?

Reply:
ah, my mind read it simply as "new [] {".  Can we switch to that as the vast majority of IDE code uses 'var' when possible.